### PR TITLE
Add Mesh & Envoy Details dashboard

### PR DESCRIPTION
**Pull request by eduser25**
- Number and list of Namespaces on the mesh
- Number and list of Services on the mesh
- Number and list of Pods on the mesh
- Envoy:
  - Malloc data
  - Heap (virt mem) data
  - Physical (RSS?) mem data
  - CPU consumption on Envoy containers

image

- Metrics                [X]

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

File: charts/osm/grafana/dashboards/osm-mesh-envoy-details.json [hunk 1]
```diff
@@ -0,0 +1,791 @@
...
+      "options": {
+        "showLabels": false,
+        "showTime": false,
+        "sortOrder": "Descending",
+        "wrapLogMessage": false
+      },
+      "pluginVersion": "7.0.1",
+      "targets": [
+        {
+          "expr": "count by(source_namespace) (envoy_http_rq_total)",
```
Comment (eduser25) [inline charts/osm/grafana/dashboards/osm-mesh-envoy-details.json:118]: envoy_http_rq_total will not work when we have TCP-only routing. We'll figure some other metric to use probably.

File: charts/osm/grafana/dashboards/osm-mesh-envoy-details.json [hunk 2]
```diff
@@ -0,0 +1,791 @@
...
+          "format": "time_series",
+          "instant": false,
+          "interval": "",
+          "legendFormat": "",
+          "refId": "A"
+        }
+      ],
+      "timeFrom": null,
+      "timeShift": null,
+      "title": "Num NS on the Mesh",
```
Comment (shashankram) [inline charts/osm/grafana/dashboards/osm-mesh-envoy-details.json:75]:
```suggestion
      "title": "Number of Namespaces",
```

File: charts/osm/grafana/dashboards/osm-mesh-envoy-details.json [hunk 3]
```diff
@@ -0,0 +1,791 @@
...
+          "instant": true,
+          "interval": "",
+          "intervalFactor": 1,
+          "legendFormat": "",
+          "refId": "A"
+        }
+      ],
+      "timeFrom": null,
+      "timeShift": null,
+      "title": "NS on the Mesh",
```
Comment (shashankram) [inline charts/osm/grafana/dashboards/osm-mesh-envoy-details.json:130]:
```suggestion
      "title": "Namespaces",
```

File: charts/osm/grafana/dashboards/osm-mesh-envoy-details.json [hunk 4]
```diff
@@ -0,0 +1,791 @@
...
+          "interval": "",
+          "legendFormat": "Envoy {{source_namespace}} / {{source_pod_name}}",
+          "refId": "A"
+        }
+      ],
+      "thresholds": [],
+      "timeFrom": null,
+      "timeRegions": [],
+      "timeShift": null,
+      "title": "Envoy Phys Mem footprint",
```
Comment (snehachhabria) [inline charts/osm/grafana/dashboards/osm-mesh-envoy-details.json:627]:
```suggestion
      "title": "Envoy Physical Memory footprint",
```

File: charts/osm/grafana/dashboards/osm-mesh-envoy-details.json [hunk 5]
```diff
@@ -0,0 +1,791 @@
...
+      "30m",
+      "1h",
+      "2h",
+      "1d"
+    ]
+  },
+  "timezone": "",
+  "title": "Mesh and Envoy Details",
+  "uid": "PLyKJcHGz",
+  "version": 2
```
Comment (snehachhabria) [inline charts/osm/grafana/dashboards/osm-mesh-envoy-details.json:790]:
```suggestion
  "version": 1
```

**Comment by aanandr** [comment]
@eduser25 - the dashboard looks awesome. The list of services, namespaces and pods in mesh is really useful 👍 
Had some questions on the graphs
1. Can we also show the CPU, storage and mem allocation for addo Pods? This will the customer how much of resource the control plane is taking
2. For the Envoy metrics can we get the cumulative nos. across all Envoys?

**Pull request merged**